lstchain/tools/lstchain_create_irf_files.py

```diff
@@ -0,0 +1,399 @@
+            else:
+                p["mc_type"] = "diffuse"
+                # For diffuse gamma using Proton Spectra for calculating event weights
+                if particle_type == "gamma":
+                    if not self.point_like:
+                        p["target_spectrum"] = IRFDOC_PROTON_SPECTRUM
+                        self.log.debug(
+                            "Proton spectrum used as target spectrum"
+                            " for MC diffuse gamma"
+                        )
+                    else:
+                        raise ToolConfigurationError(
+                            "Diffuse MC gamma cannot be used for generating "
+                            "point_like IRFs. Use appropriate MC and IRF type."
+                        )
+
+            self.log.debug(f"Simulated {p['mc_type']} {particle_type.title()} Events:")
+
+            p["simulated_spectrum"] = PowerLaw.from_simulation(
+                p["simulation_info"], 50 * u.hour
+            )
+            p["events"]["weight"] = calculate_event_weights(
+                p["events"]["true_energy"],
+                p["target_spectrum"],
+                p["simulated_spectrum"],
+            )
+
+            for prefix in ("true", "reco"):
+                k = f"{prefix}_source_fov_offset"
+                p["events"][k] = calculate_source_fov_offset(p["events"], prefix=prefix)
+            # calculate theta / distance between reco and assumed source position
+            p["events"]["theta"] = calculate_theta(
+                p["events"],
+                assumed_source_az=p["events"]["true_az"],
+                assumed_source_alt=p["events"]["true_alt"],
+            )
+            self.log.debug(p["simulation_info"])
+
+        gammas = self.mc_particle["gamma"]["events"]
+
+        self.log.info(f"Using fixed G/H cut of {self.data_sel.fixed_gh_cut}")
+
+        gammas = self.data_sel.filter_cut(gammas)
+        gammas = self.data_sel.allowed_tels_filter(gammas)
+        gammas = self.data_sel.gh_cut(gammas)
+
+        # point_like = True for point like IRFs, False for Full Enclosure IRFs
+        if self.point_like:
+            gammas = self.data_sel.theta_cut(gammas)
+            gammas = self.data_sel.true_src_fov_offset_cut(gammas)
+
+        # Binning of parameters used in IRFs
+        true_energy_bins = self.data_bin.true_energy_bins()
+        reco_energy_bins = self.data_bin.reco_energy_bins()
+        migration_bins = self.data_bin.energy_migration_bins()
+        source_offset_bins = self.data_bin.source_offset_bins()
+
+        if self.point_like:
+            # Gammapy 0.18.2 needs offset bin centers for interpolation
+            # Using just 2 'edges' like [0.2,0.6] works fine for reading the IRF but,
+            # this workaround is necessary for further analysis using gammapy.
+            if len(self.data_bin.fov_offset_bins()) != 3:
+                self.log.critical("Offset binning is not appropriate for single offset")
```

Comment:
Critical log but you do not exit?

Reply:
Ah yes! I can instead raise a ToolConfigurationError.

Reply:
As discussed below, I don't think you should even check this. If the gammas are point like, create the appropriate single offset bin without looking at the `data_bins.fov_offset_bins`